Public/Src/Tools/Xldb.Analyzer/XldbAnalyzer.cs

```diff
@@ -122,42 +123,40 @@ public int AnalyzeEventStats()
+                var listOfStorageStats = new List<DBStorageStatsValue>();
+                ulong totalCount = 0;
+                ulong totalPayload = 0;
+
+                writer.WriteLine("Storage stats (uncompressed) of the DB.\n\n");
+
+                foreach (DBStoredTypes storageType in Enum.GetValues(typeof(DBStoredTypes)))
                 {
-                    var eventCount = dataStore.GetCountByEvent(eventId);
+                    var dbStorageStatValue = dataStore.GetCountByEvent(storageType);
 
-                    if (eventCount != null)
+                    if (dbStorageStatValue != null)
                     {
-                        foreach (var workerCount in eventCount.WorkerToCountMap)
-                        {
-                            if (workerToEventDict.TryGetValue(workerCount.Key, out var eventDict))
-                            {
-                                eventDict[eventId] = workerCount.Value;
-                            }
-                            else
-                            {
-                                var dict = new Dictionary<ExecutionEventId, int>();
-                                dict.Add(eventId, workerCount.Value);
-                                workerToEventDict.Add(workerCount.Key, dict);
-                            }
-                        }
+                        totalCount += dbStorageStatValue.Count;
+                        totalPayload += dbStorageStatValue.Size;
+                        listOfStorageStats.Add(dbStorageStatValue);
+                    }
+                    else
+                    {
+                        listOfStorageStats.Add(new DBStorageStatsValue());
                     }
                 }
 
-                var maxLength = Enum.GetValues(typeof(ExecutionEventId)).Cast<ExecutionEventId>().Select(e => e.ToString().Length).Max();
-
-                foreach (var workerDict in workerToEventDict)
+                for (var i = 0; i < listOfStorageStats.Count(); i++)
                 {
-                    writer.WriteLine("Worker {0}", workerDict.Key);
-                    foreach (var eventStats in workerDict.Value)
-                    {
-                        writer.WriteLine(
-                        "{0}: Count = {1}",
-                        eventStats.Key.ToString().PadRight(maxLength, ' '),
-                        eventStats.Value.ToString(CultureInfo.InvariantCulture).PadLeft(12, ' '));
-                    }
-                    writer.WriteLine();
+                    var currDbStorageStatValue = listOfStorageStats.ElementAt(i);
+
+                    writer.WriteLine(
+                    "{0}: Count = {1} ({2} %),   Payload = {3} ({4} %)",
+                    ((DBStoredTypes)i).ToString().PadRight(maxLength, ' '),
+                    currDbStorageStatValue.Count.ToString(CultureInfo.InvariantCulture).PadLeft(15, ' '),

```

Comment:
>15 [](start = 96, length = 2)

where are the padding values coming from?

Reply:
Based on sizes of the count/payload values. Ive made them constants now.

---
In reply to: [321866040](https://github.com/microsoft/BuildXL/pull/836#discussion_r321866040) [](ancestors = 321866040)